### PR TITLE
Add support for api_token

### PR DESCRIPTION
**Pull request by krombel**
With api_token you are able to generate access to the api which has limited capabilities.
Therefore we wanted to use it and implemented it in our environment.

### CODE REVIEW

File: cloudflare.py
Comment (willbuckner) [inline cloudflare.py]: You have a typo here :) `slef.auth_key` --> `self.auth_key`

File: cloudflare.py
Comment (willbuckner) [inline cloudflare.py]: IMO, this should use `"".format()` for consistency with the rest of the codebase, rather than concatenation.

File: cloudflare.py
Comment (krombel) [reply cloudflare.py]: I don't know where that comes from...
That mistake is not in the file we use in our env...

File: cloudflare.py
Comment (krombel) [reply cloudflare.py]: done

File: cloudflare.py
Comment (sinkingpoint) [inline cloudflare.py]: Just a nitpick: 
To me this would read better as 

```
if not self.api_token and not (self.auth_email and self.auth_key):
```

But that might just be personal preference

Comment (krombel) [reply cloudflare.py]: As you wish. I think it is more straight forward to understand as well.

**Comment by bobrik** [comment]
@willbuckner, @sinkingpoint, can you approve the PR if you're happy with it?

@krombel, feel free squash when you have an approval.

File: cloudflare.py
Comment (willbuckner) [inline cloudflare.py]: nit: Sorry, I should have caught this earlier, but there's an extra space after the `else` here.

```suggestion
        else:
```

**Review by willbuckner — APPROVED** [review]

**Comment by krombel** [comment]
Thanks for your review. As you approved the PR I now squashed the commits as mentioned

**Comment by bobrik** [comment]
Thanks!

**Pull request merged**
